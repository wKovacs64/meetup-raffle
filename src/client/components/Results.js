import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import Winner from './Winner';

const Results = ({ winners, ...props }) => (
  <Fragment>
    {!!winners.length && (
      <div
        className="flex flex-wrap justify-between justify-around-ns"
        {...props}
      >
        {winners.map(winner => (
          <Winner key={winner.profileURL} winner={winner} />
        ))}
      </div>
    )}
  </Fragment>
);

Results.propTypes = {
  winners: PropTypes.arrayOf(
    PropTypes.shape({
      name: PropTypes.string.isRequired,
      photoURL: PropTypes.string.isRequired,
      profileURL: PropTypes.string.isRequired,
    }),
  ),
};

Results.defaultProps = {
  winners: [],
};

export default Results;
