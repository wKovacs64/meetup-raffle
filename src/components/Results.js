import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import { Winner } from '.';

const Results = ({ winners }) => (
  <Fragment>
    {!!winners.length && (
      <div className="flex flex-wrap justify-between justify-around-ns">
        {winners.map(winner => (
          <Winner key={winner.profileURL} winner={winner} />
        ))}
      </div>
    )}
  </Fragment>
);

Results.propTypes = {
  winners: PropTypes.arrayOf(
    PropTypes.shape({
      name: PropTypes.string.isRequired,
      photoURL: PropTypes.string.isRequired,
      profileURL: PropTypes.string.isRequired,
    }),
  ),
};

Results.defaultProps = {
  winners: [],
};

export default Results;
