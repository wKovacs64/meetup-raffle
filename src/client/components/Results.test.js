import React from 'react';
import { render } from 'react-testing-library';
import Results from './Results';

jest.mock('./Winner', () =>
  jest.fn(({ winner }) => <span>{winner.name}</span>),
);

const winners = Array.from(Array(2), (_, idx) => ({
  name: `Pickle Rick ${idx}`,
  photoURL: `https://i.imgur.com/3VhMoBD.png?i=${idx}`,
  profileURL: `https://en.wikipedia.org/wiki/Pickle_Rick?i=${idx}`,
}));

describe('Results', () => {
  const { container, getByText } = render(<Results winners={winners} />);

  it('renders', () => {
    expect(container).toMatchSnapshot();
  });

  it('contains all winners', () => {
    winners.forEach(winner => {
      expect(() => getByText(winner.name)).not.toThrow();
    });
  });
});
