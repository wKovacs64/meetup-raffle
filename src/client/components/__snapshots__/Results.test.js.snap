// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Results renders 1`] = `
<div>
  <div
    class="flex flex-wrap justify-between justify-around-ns"
  >
    <span>
      Pickle Rick 0
    </span>
    <span>
      Pickle Rick 1
    </span>
  </div>
</div>
`;
